spyder/plugins/editor/widgets/completion.py

```diff
@@ -347,14 +349,25 @@ def is_up_to_date(self, item=None):
             filter_text = completion
         return self.check_can_complete(filter_text, current_word)
 
-    def check_can_complete(self, filter_text, current_word):
+    def check_can_complete(self, filter_text, current_word, params=None):
         """Check if current_word matches filter_text."""
         if not filter_text:
             return True
+
         if not current_word:
             return True
-        if not filter_text[0].isalpha():
-            current_word = filter_text[0] + current_word
+
+        # https://github.com/spyder-ide/spyder/commit/96ff608af2059865f74c225c065c0102a4da3f45#diff-a020df4c2a52279746bacc37a5fc8ce7R332
```

Comment:
@metalogical @dalthviz @andfoy what was the purpose of this code

```python
        if not filter_text[0].isalpha():
            current_word = filter_text[0] + current_word
```

That is currently breaking completions for variables starting on underscore `_`

I would like to understand what is/was the use case so I can fix around it. I commented the code and added some tests.

Thanks

Reply:
I am not sure if this is new or not
https://github.com/spyder-ide/spyder/commit/96ff608af2059865f74c225c065c0102a4da3f45#diff-a020df4c2a52279746bacc37a5fc8ce7R332

Reply:
This existed before my changes (I just renamed the variables for clarity), and I avoided changing it since I wasn't sure either.

It doesn't make much sense to me, so I'm :+1: on removing it.

cc @ccordoba12